### DISCUSSION TITLE
Query partition parquet

### DISCUSSION THREAD

**Discussion by ac4wong** [duckdb/duckdb#4830] [Q&A] [answered]
Hi I saw that this feature was added via https://github.com/duckdb/duckdb/pull/4211

Are there any documentations or examples on how to actually make use of it?

**Comment by Alex-Monahan** [comment, marked as answer]
Howdy! It is not in the docs yet. However, in that PR there are some tests that you can use as examples! I've found the tests to be a great starting point when trying out a very recently added DuckDB feature.

https://github.com/samansmink/duckdb/blob/0fe3e43ebccfb52b7dc63f9ddb3c34f1b66a5b3c/test/sql/copy/parquet/parquet_hive.test

**Reply by ac4wong** [reply]
Thanks! That solves half of my problem.  The examples seem to focus on a single partitioned file test.parquet. 
What if I want to want to query multiple partitioned files? is that possible?


e.g. currently I have multiple files (price2018.parquet, price2019.parquet...etc) and I query them with SELECT * FROM read_parquet('price*.parquet')

If I start partitioning each of them (i.e. under price2018.parquet folder it will be partitioned by companyname), how can I write the query?

Stealing a similar example online:
![image](https://user-images.githubusercontent.com/107779561/193163950-64efa362-65af-463a-a558-cb2a2c036b18.png)

**Reply by Alex-Monahan** [reply]
Untested, but multiple wildcards are allowed:
```sql
Select * from read_parquet('price*/*.parquet')
```

**Comment by ac4wong** [comment]
thanks for the quick reply but I cannot get any of these to work to be honest.  I guess I will have to try other packages until the duckdb beef up the documentation and guides.

**Reply by Alex-Monahan** [reply]
Sorry to hear that! I believe you are quite close. Could you provide what you have tried so that I can continue to help?

**Comment by ac4wong** [comment]
So I tried to follow the test case but these are the errors I am getting.  I guess what it will be really helpful is if I can see the directory structure that was used in the testcase.
My parquet store is definitely not corrupted or anything, it works if I use Dask to read.


![image](https://user-images.githubusercontent.com/107779561/193184811-96b14fbe-ecc3-4fd1-bfd7-ba3c12c76dc2.png)
![image](https://user-images.githubusercontent.com/107779561/193184849-26e567be-6d11-44b2-8bf5-b8068c660c37.png)
![image](https://user-images.githubusercontent.com/107779561/193184888-2ff2d53b-50ad-4d56-9af9-fcc35bda5fba.png)

**Reply by Alex-Monahan** [reply]
Thanks for the details! So there are parquet files in those FactorName= folders? If so, maybe try this?
```sql
select * from parquet_scan('Z:/_TempUsage/*/factor_library_parquet/*/*/*/*.parquet', HIVE_PARTITIONING=1)
```

**Reply by ac4wong** [reply]
oh thanks it's working! I think I understand the syntax now, basically we are still treating the partition files as collection of normal parquet files.
Final question:  is parquet_scan the same as read_parquet?  Both seem to work but are they doing something different under the hood?
